pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -133,18 +133,9 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         PrecisionT result{0.0};
 
         if (obs.getObsName().find("SparseHamiltonian") != std::string::npos) {
-            // SparseHamiltonian does not support samples in pennylane.
-            PL_ABORT("For SparseHamiltonian Observables, expval calculation is "
-                     "not supported by shots");
-        } else if (obs.getObsName().find("Hermitian") != std::string::npos) {
-            // TODO support. This support requires an additional method to solve
-            // eigenpair and unitary matrices, and the results of eigenpair and
-            // unitary matrices data need to be added to the Hermitian class and
-            // public methods are need to access eigen values. Note the
-            // assumption that eigen values are -1 and 1 in the
-            // `measurement_with_sample` method should be updated as well.
-            PL_ABORT("For Hermitian Observables, expval calculation is not "
```

Comment:
Is this just caught in Observables now?

Reply:
Yea, it is caught by the `applyInPlaceShots()` in Observables.